lnwallet/btcwallet/btcwallet.go

```diff
@@ -1581,35 +1582,60 @@ func (b *BtcWallet) ListTransactionDetails(startHeight, endHeight int32,
 			currentHeight, blockPackage, b.netParams,
 		)
 		if err != nil {
-			return nil, err
+			return nil, 0, 0, err
 		}
 
 		txDetails = append(txDetails, details...)
 	}
 	for _, tx := range txns.UnminedTransactions {
 		detail, err := unminedTransactionsToDetail(tx, b.netParams)
 		if err != nil {
-			return nil, err
+			return nil, 0, 0, err
 		}
 
 		txDetails = append(txDetails, detail)
 	}
 
+	var (
+		firstIdx = indexOffset + 1
+		lastIdx  = int(indexOffset) + maxTransactions
+	)
+
 	// Return empty transaction list, if offset is more than all
 	// transactions.
 	if int(indexOffset) >= len(txDetails) {
-		return []*lnwallet.TransactionDetail{}, nil
+		firstIdx = 0
+		lastIdx = 0
+		txDetails = []*lnwallet.TransactionDetail{}
+
+		return txDetails, uint64(firstIdx), uint64(lastIdx), nil
 	}
 
-	end := int(indexOffset) + maxTransactons
+	// If maxTransactions is set to -1, then we'll return all transactions
+	// starting from the offset.
+	if maxTransactions == -1 {
+		lastIdx = len(txDetails)
+		txDetails = txDetails[indexOffset:]
+
+		return txDetails, uint64(firstIdx), uint64(lastIdx), nil
+	}
+
+	// If maxTransactions is set to a value less than -1, we should return
+	// an error.
+	if maxTransactions < -1 {
+		return nil, 0, 0, fmt.Errorf("maxTransactions must be " +
+			"equal to or greater than -1")
+	}
+
+	end := int(indexOffset) + maxTransactions
 	if end > len(txDetails) {
 		end = len(txDetails)
+		lastIdx = len(txDetails)
 	}
-	if maxTransactons == -1 {
-		return txDetails[indexOffset:], nil
-	}
 
-	return txDetails[indexOffset:end], nil
+	txDetails = txDetails[indexOffset:end]
+
+	return txDetails, uint64(firstIdx), uint64(lastIdx), nil
```

Comment:
Why do we need these two extra variables at all? Shouldn't `firstIdx = indexOffset` and `lastIdx = end-1`?

Reply:
If my [previous comment](https://github.com/lightningnetwork/lnd/pull/8998/commits/8fe74aaf32d0e1a436ba6d3ca0ecb2dd90a03ba8#r1835676617) is correct, then this wouldn't be the case. However, I can replace the `end` variable with `lastIdx`, since they are the same.

Reply:
We'll want zero-based indexes here. See my other comment, unfortunately the invoices template you looked at has a special case with 0 that we don't need to port over to this.

Reply:
I updated it to use a zero-based index.